Source/core/DataElement.h

```diff
@@ -167,12 +184,22 @@ namespace Core {
         }
         inline DataElement(const ProxyType<DataStore>& buffer, const uint64_t offset, const uint64_t size = 0)
             : m_Storage(buffer)
-            , m_Buffer(&(buffer->Buffer())[offset])
+            , m_Buffer(buffer->Size() > offset ? &(buffer->Buffer())[offset] : nullptr)
```

Comment:
This is for example additional code where later on you will actually (using defensive coding) ASSERT on this. No need to than check it here as well. so This change is additional code for usage outside of the working area of this software AND this is validated later on using an ASSERT. So the ASSERT is great, which makes the additional code here unnecessary.